plugins/controlboard/src/ControlBoard.cpp

```
@@ -241,10 +260,27 @@ bool ControlBoard::setJointProperties(EntityComponentManager& _ecm)
             gzJoint.EnableTransmittedWrenchCheck(_ecm, true);
 
             // Initialize JointProperties object
-            m_controlBoardData.joints[i].name = jointFromConfigName;
+            m_controlBoardData.physicalJoints[i].commonJointProperties.name = jointFromConfigName;
 
             yInfo() << "Joint " << jointFromConfigName << " added to the control board data.";
         }
+        // Let's initialize all the buffers/vectors
+        configureBuffers();
```

This function can fail, can we propagate the error to the caller?